framework/Models/EnsembleModel.py

```diff
@@ -697,42 +652,33 @@ def _externalRun(self,inRun, jobHandler):
               if modelToRemove != modelIn:
                 jobHandler.getFinished(jobIdentifier = modelToRemove + utils.returnIdSeparator() + identifier, uniqueHandler = self.name + identifier)
             self.raiseAnError(RuntimeError,"The Model  " + modelIn + " identified by " + finishedRun[0].identifier +" failed!")
-
-          # collect output in the temporary data object
-          #exportDict = self.modelsDictionary[modelIn]['Instance'].collectOutput(finishedRun[0], True)
-          #else:
-          #  exportDict = holdCollector[modelIn]['exportDict']
           # store the output dictionary
           tempOutputs[modelIn] = copy.deepcopy(evaluation)
           # collect the target evaluation
           #if modelIn not in modelsOnHold:
-          self.modelsDictionary[modelIn]['Instance'].collectOutput(finishedRun[0],tempTargetEvaluations[modelIn])
+          self.modelsDictionary[modelIn]['Instance'].collectOutput(finishedRun[0],inRunTargetEvaluations[modelIn])
           ## FIXME: The call asDataset() is unuseful here. It must be done because otherwise the realization(...) method from collector
           ## does not return the indexes values (TO FIX)
-          tempTargetEvaluations[modelIn].asDataset()
+          inRunTargetEvaluations[modelIn].asDataset()
           # get realization
-          dataSet = tempTargetEvaluations[modelIn].realization(index=0,unpackXArray=True)
+          dataSet = inRunTargetEvaluations[modelIn].realization(index=iterationCount-1,unpackXArray=True)
           ##FIXME: the following dict construction is a temporary solution since the realization method returns scalars if we have a PointSet
           dataSet = {key:np.atleast_1d(dataSet[key]) for key in dataSet}
```

Comment:
I think this might be the line that causes a user to need to change their externalModel python code in order to use it in an ensemble model. Without an ensemble model, `self.myScalar` is a float, while in the ensemble model, `self.myScalar` is a length-1 numpy array.  How helpful is it to force the 1-length arrays?

Reply:
This should have been addressed (in the restructuring done in the past for the dataobject) in the ```__retrieveDependentOutput```:
```dependentOutputs[inKey] =  previousOutputs[inKey] if len(previousOutputs[inKey]) > 1 else previousOutputs[inKey][0]```